code/modules/loadout/loadout_items.dm

```
@@ -0,0 +1,348 @@
+		// Otherwise set this automatically to true if it is actually colorable
+		can_be_greyscale = TRUE
+
+	if(isnull(name))
+		name = item_path::name
+
+	if(GLOB.all_loadout_datums[item_path])
+		stack_trace("Loadout datum collision detected! [item_path] is shared between multiple loadout datums.")
+	GLOB.all_loadout_datums[item_path] = src
+
+	if(isnull(ui_icon) && isnull(ui_icon_state))
+		ui_icon = item_path::icon_preview || item_path::icon
+		ui_icon_state = item_path::icon_state_preview || item_path::icon_state
+
+	if(can_be_reskinned)
+		var/obj/item/dummy_item = new item_path()
+		if(!length(dummy_item.unique_reskin))
+			can_be_reskinned = FALSE
+			stack_trace("Loadout item [item_path] has can_be_reskinned set to TRUE but has no unique reskins.")
+		else
+			cached_reskin_options = dummy_item.unique_reskin.Copy()
+		qdel(dummy_item)
+
+/datum/loadout_item/Destroy(force, ...)
+	if(force)
+		stack_trace("QDEL called on loadout item [type]. This shouldn't ever happen. (Use FORCE if necessary.)")
+		return QDEL_HINT_LETMELIVE
+
+	GLOB.all_loadout_datums -= item_path
+	return ..()
+
+/**
+ * Takes in an action from a loadout manager and applies it
+ *
+ * Useful for subtypes of loadout items with unique actions
+ *
+ * Return TRUE to force an update to the UI / character preview
+ */
+/datum/loadout_item/proc/handle_loadout_action(datum/preference_middleware/loadout/manager, mob/user, action, params)
+	SHOULD_CALL_PARENT(TRUE)
+
+	switch(action)
+		if("select_color")
+			if(can_be_greyscale)
+				return set_item_color(manager, user)
+
+		if("set_name")
+			if(can_be_named)
+				return set_name(manager, user)
+
+		if("set_skin")
+			return set_skin(manager, user, params)
+
+	return FALSE
+
+/// Opens up the GAGS editing menu for a certain item
+/datum/loadout_item/proc/set_item_color(datum/preference_middleware/loadout/manager, mob/user)
+	if(manager.menu)
+		tgui_alert(user, "You already have a color menu open!")
```

I think it should either silently fail or close the last menu because otherwise if you're clicking the color prompt a couple times because it's the server initializing and it's laggy, you're just gonna get more annoyed with all the popups.